OpenRA.Mods.Cnc/Activities/Infiltrate.cs

```diff
@@ -71,6 +71,9 @@ protected override void OnEnterComplete(Actor self, Actor targetActor)
 				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech",
 					infiltrates.Info.Notification, self.Owner.Faction.InternalName);
 
+			if (self.Owner == self.World.LocalPlayer)
```

Comment:
This pattern is used for nearly all calls to make sure that notifications are only displayed for active players (and not spectators). Perhaps it should be moved in the method to avoid repeating it? Something like:
```cs
public static void AddTransientLine(string text, Player player)
{
	if (string.IsNullOrEmpty(text))
		return;

	// null check is to allow notifications to everyone
	// so it matches speech notifications via Sound.PlayPredefined
	if (player == player.World.LocalPlayer || player == null)
		AddTextNotification(TextNotificationPool.Transients, SystemMessageLabel, text);
}
```